### PR TITLE
libzip: fix package tests

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix `libzip` package tests

###### Change Log  

- Fix `libzip` package tests

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: [link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=634189&view=results) [1](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=633694&view=results) [2](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=624683&view=results)

### CODE REVIEW

File: SPECS/libzip/libzip.spec [hunk 1]
```diff
@@ -106,6 +113,10 @@ make check
 %{_libdir}/pkgconfig/libzip.pc
 
 %changelog
+* Mon Aug 19 2024 Andrew Phelps <anphel@microsoft.com> - 1.10.1-2
```
Comment (mfrw) [inline SPECS/libzip/libzip.spec:116]: The test build has a failure in a very weird way:

```bash
INFO[0072][scheduler] Passed SRPMs tests:                          
INFO[0072][scheduler] --> libzip-1.10.1-2.azl3.src.rpm             
INFO[0072][scheduler] Failed SRPMs:                                
INFO[0072][scheduler] --> libzip-1.10.1-2.azl3.src.rpm , error: exit status 2, for details see: /mnt/vss/_work/1/s/build/logs/pkggen/rpmbuilding/libzip-1.10.1-2.azl3.src.rpm.log 
```

Comment (anphel31) [reply SPECS/libzip/libzip.spec:116]: Yep, it looks like the build is passing when the check tests are enabled. But it fails when running without the tests with:
`"ERROR: No matching distribution found for nihtest"`
`nihtest` is not currently available in our specs, so it is installed using pip. But networking is only enabled when running tests, so pip cannot install during non-test builds.

Comment (anphel31) [reply SPECS/libzip/libzip.spec:118]: I've resolved this by adding a patch to modify CMakeLists.txt to skip the part where it disables regression tests when nihtest is not installed. This allows the tests to run, but also allows building without the tests.

**Review by sameluch — APPROVED** [review]
Looks Good!

**Pull request merged**